client/packages/invoices/src/Prescriptions/LineEditView/PrescriptionLineEditForm.tsx

```diff
@@ -89,396 +89,394 @@ export const PrescriptionLineEditForm: React.FC<
 
-    const packSize =
-      packSizeController.selected?.value !== -1
-        ? (packSizeController.selected?.value ?? 1)
-        : 1;
+      const quantity = inputUnitQuantity === undefined ? 0 : inputUnitQuantity;
+      setIssueUnitQuantity(quantity);
 
-    const numPacks = quantity / packSize;
-    debouncedAllocate(
-      numPacks,
-      Number(packSize),
-      quantityType === 'prescribed' ? inputUnitQuantity : prescribedQuantity
-    );
-  };
+      const packSize =
+        packSizeController.selected?.value !== -1
+          ? (packSizeController.selected?.value ?? 1)
+          : 1;
 
-  const handlePrescribedQuantityChange = (inputPrescribedQuantity?: number) => {
-    if (inputPrescribedQuantity == null) return;
-    setPrescribedQuantity(inputPrescribedQuantity);
-    handleIssueQuantityChange(inputPrescribedQuantity, 'prescribed');
-  };
+      const numPacks = quantity / packSize;
+      debouncedAllocate(
+        numPacks,
+        Number(packSize),
+        quantityType === 'prescribed' ? inputUnitQuantity : prescribedQuantity
+      );
+    };
 
-  const prescriptionLineWithNote = draftPrescriptionLines.find(l => !!l.note);
-  const note = prescriptionLineWithNote?.note ?? '';
+    const handlePrescribedQuantityChange = (inputPrescribedQuantity?: number) => {
+      if (inputPrescribedQuantity == null) return;
+      setPrescribedQuantity(inputPrescribedQuantity);
+      handleIssueQuantityChange(inputPrescribedQuantity, 'prescribed');
+    };
 
-  useEffect(() => {
-    if (preferences?.editPrescribedQuantityOnPrescription) {
-      const selectedItem = items.find(
-        prescriptionItem => prescriptionItem.id === item?.id
-      );
-      const newPrescribedQuantity =
-        selectedItem?.lines?.find(
-          ({ prescribedQuantity }) => prescribedQuantity != null
-        )?.prescribedQuantity ?? 0;
-      setPrescribedQuantity(newPrescribedQuantity);
-    }
+    const prescriptionLineWithNote = draftPrescriptionLines.find(l => !!l.note);
+    const note = prescriptionLineWithNote?.note ?? '';
 
-    const newIssueQuantity = Math.round(
-      allocatedUnits / Math.abs(Number(packSizeController.selected?.value || 1))
-    );
-    if (newIssueQuantity !== issueUnitQuantity)
-      setIssueUnitQuantity(newIssueQuantity);
-    setAllocationAlerts([]);
+    useEffect(() => {
+      if (preferences?.editPrescribedQuantityOnPrescription) {
+        const selectedItem = items.find(
+          prescriptionItem => prescriptionItem.id === item?.id
+        );
+        const newPrescribedQuantity =
+          selectedItem?.lines?.find(
+            ({ prescribedQuantity }) => prescribedQuantity != null
+          )?.prescribedQuantity ?? 0;
+        setPrescribedQuantity(newPrescribedQuantity);
+      }
 
-    setAbbreviation('');
-    setDefaultDirection('');
-  }, [item?.id]);
+      const newIssueQuantity = NumUtils.round(allocatedUnits / Math.abs(Number(packSizeController.selected?.value || 1)), 2);
```

Comment:
Rounding to 2dp here also to be consistent with other parts of app for input values.